### ISSUE TITLE
overheating

### ISSUE THREAD

**Issue by heisenberg-63** [eranif/codelite#2487]
by working on  latest codelite my Macbook air fans have to overspeed as it heats up the Mac

**Comment by eranif** [comment]
Do you get anything on your `~/Library/Application Support/codelite/codelite.log`?
Can you put this in `tail` ?

**Comment by TaiSHiNet** [comment]
Same thing is happening to me, nothing abnormal in the log:
`[12:43:53:730 ERR] execvp(' brew list llvm ') error: Invalid argument`

Although Activity Monitor is showing the CodeLite child thread pinning ~100% CPU

**Comment by eranif** [comment]
@TaiSHiNet the above error is an attempt of CodeLite to locate clang installation using `brew`. You don't have it installed (or at least it is not in the path) so this error is OK.

Can you increase the log verbosity from: Settings -> Preferences -> Misc -> Log -> CodeLite log verbosity -> Debug

and `tail` the log file?

**Comment by TaiSHiNet** [comment]
@eranif https://gist.github.com/TaiSHiNet/85d2654aeab903ae67d2495d1cbcc573

**Comment by blockchaindevel** [comment]
I was just going to report this as well and found this open issue.
Same problem - CPU pinning at close to 100%
MacBook Pro running OS X 10.15.4 and CodeLite 14.0.0. 
----
Can someone check the following on your end:
Go to Plugins->Manage Plugins and Unload the Language Server Plugin.
CPU drops way down!
Load it again and it pins at near 100%
Confirmed on different Macs

**Comment by eranif** [comment]
@blockchaindevel from plugins -> language server settings
Can you write which LSPs are configured for you?

**Comment by blockchaindevel** [comment]
@eranif none seem to be configured.

<img width="841" alt="Screen Shot 2020-05-07 at 8 55 15 AM" src="https://user-images.githubusercontent.com/52295768/81297400-2ec29700-9041-11ea-8ec1-949b18058611.png">

**Comment by eranif** [comment]
can you try and disable plugins, I would start with:

```
WebTools
Subversion
```

**Comment by eranif** [comment]
You can also try starting CodeLite without plugins at all from a terminal:

```
open /path/to/codelite.app/ --args --no-plugins
```

**Comment by blockchaindevel** [comment]
@eranif 
Started CodeLite with no plugins at all -- CPU usage low.
Once I enable ONLY LSP plugin it pins at near 100%

Same message as mentioned in log by others:

 [17:04:27:458 ERR] execvp(' brew list llvm ') error: Invalid argument

What are you seeing on your end?

**Comment by eranif** [comment]
Is this the *last* log message that you have in the log file?
can you please set log level to debug?

**Comment by eranif** [comment]
or even to Developer?

**Comment by eranif** [comment]
There are 2 more options:
Run codelite under `dtruss`:

```
sudo dtruss codelite.app/Contents/MacOS/codelite
```

and see why it consumes so much CPU

OR, run CodeLite under `lldb` or `gdb`
and when it starts to spin 100% cpu hit Ctrl-C in the debugger terminal and sample the backtrace. by typing `bt` and send me the backtraces

**Comment by blockchaindevel** [comment]
Log level was already set to Debug.
It's the next to last message. The last message is:
[Main] Keyboard Manager: Initializing keyboard manager

Are you unable to reproduce the issue on your end?

**Comment by eranif** [comment]
No, I am unable to reproduce this.
I am using MacBook Pro with OSX version 10.14.6 (Mojave)

**Comment by eranif** [comment]
I upgraded my mac to 10.15.4 today and rebuild CodeLite.
I tested it locally and when idle it consume about 6-7%. 
I will be releasing new version tomorrow. Please give it a try and let me.

My suspicious is the new linking rules Apple added (for example: non signed application can't link against libcrypto etc, but CodeLite does...) that are the root cause of the problem.

I will ping here when the new version is out so you can try it out

**Comment by eranif** [comment]
FYI: I released CodeLite 14.0.1. For macOS, this version is built on macOS 10.15.4 (Catalina)
Tested locally and it does not consume CPU. Let me know if it helps

**Comment by blockchaindevel** [comment]
Just upgraded to 14.0.1 and it looks good. CPU usage nice and low.
Thanks!

**Comment by eranif** [comment]
thanks for checking and getting back to me

**Comment by TaiSHiNet** [comment]
I know the issue is closed, but tested 14.0.1 as well, works like a dream.
Thanks for fixing it @eranif!

**Comment by Devlonoah** [comment]
Disabling the rust language server did the trick for me, CPU usage dropped from 100.2 % to 3 (approximately)

<img width="679" alt="Screenshot 2024-08-13 at 1 45 18 PM" src="https://github.com/user-attachments/assets/664d7944-649b-4656-825b-fe17ac20b0b5">

<img width="1182" alt="Screenshot 2024-08-13 at 1 45 28 PM" src="https://github.com/user-attachments/assets/af3b81db-72cd-4b3e-8614-b0d2845d71a6">
